dojo/utils.py

```diff
@@ -266,6 +266,11 @@ def deduplicate_uid_or_hash_code(new_finding):
 
 
 def set_duplicate(new_finding, existing_finding):
+    if existing_finding.test.engagement != new_finding.test.engagement:
```

Comment:
i'm confused with this portion of code. the debug log added states that we're skipping dedupe but we're not skipping dedupe, we're just adding a log
moreover, the deduplication on engagement mismatch verification is already done before calling the set_duplicate function and set_duplicate is not called in case of mismatch

Reply:
It was just intended as a safeguard as I thought there were maybe more present or future locations from where functionality could accidentally set mark a finding as a duplicate from another engagement. For example the similar finding functionality, but it turns out that it has its own logic in its views.py.

The code does stop the marking of the finding as a duplicate, so the log is correct? Maybe better wording needed if it's not clear. 

Or just remove the check altogether and trust future contributors to perform a check engagement matching before calling this method.

Reply:
maybe i'm missing something but i'm only seeing two if statements that lead to a log. apart from this log this isn't doing anything in the set_duplicate function, is it?

Reply:
no matter it those `if` are true or not, the rest of the function will be executed

Reply:
ah, you're right there's missing a 'return' or 'else'. Will update tomorrow.

Reply:
if right now this is correcly handled before calling the function, i don't think it's a good idea to add another verification in this function as it'll add noise and make the code harder to understand. as the function name states, this was intended to just mark as duplicate; not do some verifications to see if we should or not mark as duplicate :)

What can be done is add a comment about what this function is doing and that it should only be called when we're sure we want to mark as duplicate and there is no engagement mismatch. what do you think?

Reply:
agreed, I was already in doubt when I was adding this check. I have removed it. I did add it to finding/views.py as it seems that views are the place in django where input validation can/should take place. let me know what you think.